### DISCUSSION TITLE
config confusion.

### DISCUSSION THREAD

**Discussion by tmdag** [hrkfdn/ncspot#1465] [Q&A]
I have installed ncspot on fedora 39 via flatpak.

I wanted to modify config for my needs, but here is where I am little bit lost:
On the main github page, we have "A configuration file can be provided. The default location is ~/.config/ncspot."
But in the docs "https://github.com/hrkfdn/ncspot/blob/main/doc/users.md" it states that there should be "config.toml" in user directory, and you can figure it out, where it is by running "ncspot info".

after digging through confusing flatpack, i did run 
```
$ /var/lib/flatpak/app/io.github.hrkfdn.ncspot/current/active/files/bin >  ./ncspot info
USER_CONFIGURATION_PATH /home/user/.config/ncspot
USER_CACHE_PATH /home/user/.cache/ncspot
USER_RUNTIME_PATH /run/user/XXXXXXXXX/ncspot
```
So we have this file " /home/user/.config/ncspot" which does not exist and does not have .toml extension.

As it is empty, I have no idea what formatting i should be providing here.

Also, As I have already logged in before, I was expecting to see my login information within this config, but yet its empty.

**Comment by ThomasFrans** [comment]
The config file doesn't hold login info. It's also not automatically generated. You can create one under the directory shown by `ncspot info`. It will then be used the next time you start.

**Reply by tmdag** [reply]
Thanks Thomas, but as I wrote, it is not clear what format this config is meant to be  is it TOML formatting? (I'm just guessing here based on the mention about "config.toml" file that does not exist anywhere )

I can see the path after running `ncspot info` (as you can see on my post) but I don't see ncspot reacting to anything i add to it, hence I have no idea whenever maybe my config formatting is just wrong.
Might be simpler to generate config with hashed out potential options for the user to turn on, that way it would be clear for everyone.

> The config file doesn't hold login info.

Somewhere this login info is being held as I am automatically logged in each time i open ncspot, even though file under the  USER_CONFIGURATION_PATH  is empty

**Reply by ThomasFrans** [reply]
My config as reference: https://codeberg.org/ThomasFrans/dotfiles/src/commit/693188b6eaa0ffd7c53a4e1daf4e0b0946629d0e/.config/ncspot/config.toml

Also, login info isn't meant to be edited by the user. I believe there is a logout command to log out if needed. It's stored in a binary format from what I remember.

**Reply by tmdag** [reply]
Thank you Thomas! 
So we got one piece of a puzzle here. Now I know now that the config is supposed to be indeed in a toml format. 

Unfortunately, applying example config does nothing to the app. And still, there is some confusion with documentation here:
```
./ncspot info
USER_CONFIGURATION_PATH /home/user/.config/ncspot
```
is '/home/user/.config/ncspot' a file or is it a folder ncspot where config.toml should be placed? 
as per doc and suggested info, 'ncspot' IS a config file, but as per other places in docs and your example file - this is a folder where config.toml resides.

Either way, tested both, none of them did anything to the app

**Reply by Bettehem** [reply]
`~/.config/ncspot` is a directory. You should place your `config.toml` inside it.
